Libraries/LibWeb/HTML/HTMLOptionsCollection.cpp

```diff
@@ -104,7 +104,7 @@ WebIDL::ExceptionOr<void> HTMLOptionsCollection::set_value_of_indexed_property(u
 
         // 4. If n is greater than zero, then append a DocumentFragment consisting of n-1 new option elements with no attributes and no child nodes to the select element on which the HTMLOptionsCollection is rooted.
         if (n > 0) {
-            for (WebIDL::UnsignedLong i = 0; i < n - 1; i++) {
+            for (WebIDL::UnsignedLong i = 0; i <= n - 1; i++) {
```

Comment:
This does not match the spec. For example, if `n == 1`, we should append `n - 1 -> 0` new option elements. You're now appending a single option element instead.